### PR TITLE
Changes in Jupyter chart

### PR DESCRIPTION
**Pull request by ashetkar**
0. Using the latest docker images published on SnappyData's 
1. Included jupyter chart into the umbrella chart
2. Renamed umbrella chart as 'spark-umbrella'
3. Moved all conf directories of charts under conf/
4. Changes in Top level README to be added.
5. Tested on PKS (small) cluster.

### CODE REVIEW

File: charts/jupyter-with-spark/README.md [hunk 1]
```diff
@@ -24,33 +24,87 @@ This chart is work-in-progress
   The above command will deploy the helm chart and will display instructions to access Jupyter service and Spark UI.
   Note that the Spark UI will be available only after you start a SparkContext in Jupyter notebook.
 
-3. Jupyter server URL can be found by running following commands:
+3. Jupyter notebook server URL can be found by running the commands displayed after running `helm install ...` above. 
 
 ```bash
-  $ export JUPYTER_SERVICE_IP=$(kubectl get svc --namespace default jupyter-notebook-jupyter-with-spark -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
-  $ echo http://$JUPYTER_SERVICE_IP:{{ .Values.jupyterService.port }}
+  $ export JUPYTER_SERVICE_IP=$(kubectl get svc --namespace default jupyter-jupyter-notebook-ui -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
```
Comment (jramnara) [inline charts/jupyter-with-spark/README.md:30]: When including this in the top level readme .... we are simply asking the user to wait for 'kubectl get services' to provide the external IP.

File: charts/jupyter-with-spark/README.md [hunk 2]
```diff
@@ -24,33 +24,87 @@ This chart is work-in-progress
...
+   ```python
+   sparkEventLog:
+     enableHistoryEvents: true
+   ```
+ 
+   2. Specify the location of event log directory on your GCS bucket in values.yaml.
+
+   ```python
+   # eventsLogDir should point to a URI of GCS bucket where history events will be dumped
+   eventLogDir: "gs://my_bucket_name/event_logs/"
```
Comment (jramnara) [inline charts/jupyter-with-spark/README.md]: Lets make sure the bucket name we use is the same across the various charts.

File: charts/jupyter-with-spark/README.md [hunk 3]
```diff
@@ -24,33 +24,87 @@ This chart is work-in-progress
...
+
+Change this password by setting it as the value of `jupyterService.password` attribute in values.yaml.
+
+```python
+jupyterService:
+  type: LoadBalancer
+  port: 8888
+  # Set your password to access the notebook server. A default ('abc123') has been set for you.
+  # Setting the password to empty string will disable the authentication (not recommended).
+  password: 'your-new-password'
```
Comment (jramnara) [inline charts/jupyter-with-spark/README.md:77]: Seems like something no one will really use. Plain text passwd with no obfuscation is pretty much useless. 
If there is more to the security model, provide a link to the relevant docs.

File: charts/jupyter-with-spark/README.md [hunk 4]
```diff
@@ -24,33 +24,87 @@ This chart is work-in-progress
...
+See spark-defaults.conf under conf/spark/ directory of the chart for more details. You can also override the conf properties or set new ones in the notebook.
 
 ```python
-conf.set("spark.master", "k8s://https://{}:{}".format(
-        os.getenv("KUBERNETES_SERVICE_HOST"),
-        os.getenv("KUBERNETES_SERVICE_PORT")
-      ))
+spark = SparkSession.builder.config("spark.app.name", "spark-pi")\
+      .config("spark.kubernetes.namespace", "dev-namespace")\
+      .config("spark.executor.instances", "5")\
```
Comment (jramnara) [inline charts/jupyter-with-spark/README.md]: Can you keep this to 1 or 2 instances? else, by default, you won't have the resources to run. I have increased the PKS cluster to 5 workers, now.

**Review by jramnara — APPROVED** [review]

File: charts/jupyter-with-spark/README.md (hunk 3 above)
Comment (ashetkar) [reply charts/jupyter-with-spark/README.md:77]: Will update this to pass as a secret.

**Pull request merged**